balancer-js/src/modules/data/token-prices/provider.ts

```diff
@@ -0,0 +1,31 @@
+import type { Findable, Price } from '@/types';
+import { AaveRates } from './aave-rates';
+import { CoingeckoPriceRepository } from './coingecko';
+
+export class TokenPriceProvider implements Findable<Price> {
+  constructor(
+    private coingeckoRepository: CoingeckoPriceRepository,
+    private aaveRates: AaveRates
```

Comment:
It seems weird that this depends on AaveRates even when you're fetching prices of tokens that are unrelated to Aave. Also we may have many other aave-like providers in the future which would make this constructor have many different optional services and would be hard to use if you only need the 3rd optional service. 

I'm thinking this should take an `options` or `optionalServices` parameter that takes Aave for now and more services in the future but can work fine without them.